store/store.go

```diff
@@ -131,6 +131,22 @@ func infoFromRemote(d *snapDetails) *snap.Info {
 		info.Contact = d.SupportURL
 	}
 
+	// fill in the plug/slot data
+	if rawYamlInfo, err := snap.InfoFromSnapYaml([]byte(d.SnapYAML)); err == nil {
+		if info.Plugs == nil {
+			info.Plugs = make(map[string]*snap.PlugInfo)
```

Comment:
Note that plugs refer to apps (and vice versa). Is this code correctly linking plugs/slots/apps/hooks from one info object together? Are we mixing two disconnected objects here?

Reply:
Indeed, nice catch. I added a test and ensure now that the right *Snap is set (before it was wrong :(